scylla-cdc/src/cdc_types.rs

```diff
@@ -64,12 +69,16 @@ impl SerializeCql for StreamID {
     }
 }
 
-impl FromCqlVal<CqlValue> for StreamID {
-    fn from_cql(cql_val: CqlValue) -> Result<Self, FromCqlValError> {
-        let id = cql_val
-            .as_blob()
-            .ok_or(FromCqlValError::BadCqlType)?
-            .to_owned();
+impl<'frame, 'metadata> DeserializeValue<'frame, 'metadata> for StreamID {
+    fn type_check(typ: &ColumnType) -> Result<(), scylla::deserialize::TypeCheckError> {
+        <Vec<u8> as DeserializeValue<'frame, 'metadata>>::type_check(typ)
+    }
+
+    fn deserialize(
+        typ: &'metadata ColumnType<'metadata>,
+        v: Option<scylla::deserialize::FrameSlice<'frame>>,
+    ) -> Result<Self, scylla::deserialize::DeserializationError> {
+        let id = <Vec<u8> as DeserializeValue<'frame, 'metadata>>::deserialize(typ, v)?;
```

Comment:
@wprzytula maybe we should have an attribute for our derive macros, let's call it `transparent` that could be used for single-field structs / enum structs? It is a common pattern I see when people use the driver that they have to manually implement the traits in such case.

Reply:
:100: this is a great idea!